database/factory/factory.go

```diff
@@ -0,0 +1,93 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package factory
+
+import (
+	"fmt"
+
+	"github.com/ava-labs/avalanchego/api/metrics"
+	"github.com/ava-labs/avalanchego/database"
+	"github.com/ava-labs/avalanchego/database/corruptabledb"
+	"github.com/ava-labs/avalanchego/database/leveldb"
+	"github.com/ava-labs/avalanchego/database/memdb"
+	"github.com/ava-labs/avalanchego/database/meterdb"
+	"github.com/ava-labs/avalanchego/database/pebbledb"
+	"github.com/ava-labs/avalanchego/database/versiondb"
+	"github.com/ava-labs/avalanchego/utils/logging"
+)
+
+// New creates a new database instance based on the provided configuration.
+//
+// It also wraps the database with a corruptable DB and a meter DB.
+//
+// dbName is the name of the database, either leveldb, memdb, or pebbledb.
+// dbPath is the path to the database folder.
+// readOnly indicates if the database should be read-only.
+// dbConfig is the database configuration in JSON format.
+// dbMetricsPrefix is used to create a new metrics registerer for the database.
+// meterDBRegName is used to create a new metrics registerer for the meter DB.
+func New(
+	name string,
+	path string,
+	readOnly bool,
+	config []byte,
+	gatherer metrics.MultiGatherer,
+	logger logging.Logger,
+	metricsPrefix string,
+	meterDBRegName string,
+) (database.Database, error) {
+	dbRegisterer, err := metrics.MakeAndRegister(
+		gatherer,
+		metricsPrefix,
+	)
+	if err != nil {
+		return nil, err
```

Comment:
nit: could you create more specific formatted errors to the ones here ?
i.e. fmt.Errorf("failed to MakeAndRegister the metricPrefix %s : %w". metricsPrefix, err)

same for the others error in this function.

Reply:
I agree with this comment but I don't think it makes sense to block this PR on it because this is a strict refactor.